### PR TITLE
Improve Token, Credentials wording + store both credentials_key and token.

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

File: daybed/views/hello.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 @hello.get()
 def get_hello(request):
     """Return information regarding the current instance."""
-    return dict(token=request.token,
+    return dict(credentials_id=request.credentials_id,
```
Comment (leplatrem) [inline daybed/views/hello.py]: I would prefer

```
dict(credentials={'id': request.credentials_id}...
```

File: daybed/views/tokens.py [hunk 2]
```diff
@@ -14,11 +13,11 @@
...
     """Creates a new token and store it"""
-    session_token, credentials = get_hawk_credentials()
-    request.db.add_token(credentials["id"], credentials["key"])
+    token, credentials = get_hawk_credentials()
+    request.db.store_credentials(token, credentials)
 
     request.response.status = "201 Created"
     return {
-        'sessionToken': session_token,
+        'sessionToken': token,
```
Comment (leplatrem) [inline daybed/views/tokens.py]: I would put `'token': token`

Comment (Natim) [inline daybed/views/tokens.py]: I agree but this will change the API.

File: daybed/backends/redis/__init__.py [hunk 3]
```diff
@@ -171,19 +171,29 @@ def delete_model(self, model_id):
...
+    def store_credentials(self, token, credentials):
         # Check that the token doesn't already exist.
+        assert 'id' in credentials and 'key' in credentials
         try:
-            self.get_token(tokenHmacId)
-            raise TokenAlreadyExist(tokenHmacId)
-        except TokenNotFound:
+            self.get_token(credentials['id'])
+            raise backend_exceptions.TokenAlreadyExist(credentials['id'])
+        except backend_exceptions.TokenNotFound:
```
Comment (leplatrem) [inline daybed/backends/redis/__init__.py]: I would suggest to rename `TokenNotFound` to `CredentialsNotFound`, and `TokenAlreadyExist` to `CredentialsAlreadyExist`

File: daybed/views/hello.py (hunk 1 above)
Comment (Natim) [inline daybed/views/hello.py]: Actually I propose to remove this from the Hello page and return all the current informations on `GET /tokens` when connected. Will do that in another PR.

File: daybed/backends/redis/__init__.py (hunk 3 above)
Comment (Natim) [inline daybed/backends/redis/__init__.py]: If you want yes

File: daybed/permissions.py [hunk 4]
```diff
@@ -175,20 +172,20 @@ def __init__(self, request):
...
 
 
-def build_user_principals(token, request):
-    return [token]
+def build_user_principals(credentials_id, request):
+    return [credentials_id]
 
 
-def check_api_token(tokenId, tokenKey, request):
+def check_api_credentials(credentials_id, credentials_key, request):
```
Comment (almet) [inline daybed/permissions.py]: updated with `check_credentials` as a name.

File: daybed/backends/exceptions.py [hunk 5]
```diff
@@ -1,8 +1,8 @@
-class TokenAlreadyExist(Exception):
+class CredentialsAlreadyExist(Exception):
```
Comment (almet) [inline daybed/backends/exceptions.py:1]: It's actually the id that already exists.

File: daybed/backends/memory/__init__.py [hunk 6]
```diff
@@ -138,21 +139,29 @@ def delete_model(self, model_id):
...
         except KeyError:
-            raise TokenNotFound(tokenHmacId)
+            raise backend_exceptions.CredentialsNotFound(credentials_id)
 
-    def add_token(self, tokenHmacId, secret):
+    def get_credentials_key(self, credentials_id):
+        try:
+            return str(self._db['credentials_keys'][credentials_id])
+        except KeyError:
+            raise backend_exceptions.CredentialsNotFound(credentials_id)
```
Comment (almet) [inline daybed/backends/memory/__init__.py:152]: CredentialsId not found.

File: daybed/backends/exceptions.py (hunk 5 above)
Comment (Natim) [inline daybed/backends/exceptions.py:1]: Actually it is everything linked to that id that exists. token, and credentials_key

Comment (Natim) [inline daybed/backends/exceptions.py:1]: It is why I liked Token instead of Credentials here.

File: daybed/backends/redis/__init__.py [hunk 7]
```diff
@@ -171,19 +171,29 @@ def delete_model(self, model_id):
...
+
+    def get_credentials_key(self, credentials_id):
         """Retrieves a token by its id"""
-        secret = self._db.get("token.%s" % tokenHmacId)
-        if secret is None:
-            raise TokenNotFound(tokenHmacId)
-        return secret.decode("utf-8")
+        credentials_key = self._db.get("credentials_key.%s" % credentials_id)
+        if credentials_key is None:
+            raise backend_exceptions.CredentialsNotFound(credentials_id)
```
Comment (almet) [inline daybed/backends/redis/__init__.py:185]: CredentialsIdNotFound

File: daybed/permissions.py [hunk 8]
```diff
@@ -129,16 +126,16 @@ def permits(self, context, principals, permission):
...
                 model_permissions = {}
                 if permission != 'post_model':
                     # Prevent unauthorized error to shadow 404 responses
                     return True
             finally:
-                for perm_name, tokens in iteritems(model_permissions):
-                    # If one of the principals is in the valid tokens for this,
-                    # permission, grant the permission.
-                    if principals.intersection(tokens):
+                for perm_name, credentials_ids in iteritems(model_permissions):
```
Comment (almet) [inline daybed/permissions.py:135]: credential_ids is identifiers.

Comment (Natim) [inline daybed/permissions.py:135]: I didn't get what you mean. You prefer `credential_ids` rather than `credentials_ids`?
